stdlib/public/core/ContiguousArrayBuffer.swift

```diff
@@ -300,7 +300,22 @@ internal struct _ContiguousArrayBuffer<Element>: _ArrayBufferProtocol {
          realMinimumCapacity._builtinWordValue, Element.self)
 
       let storageAddr = UnsafeMutableRawPointer(Builtin.bridgeToRawPointer(_storage))
-      if let allocSize = _mallocSize(ofAllocation: storageAddr) {
+      
+      func _mallocSizeShim(
+        requestedByteCount: Int,
+        allocation: UnsafeMutableRawPointer
+      ) -> Int? {
+        if _fastPath(requestedByteCount <= 256) {
+          return requestedByteCount &+ 15 & ~15
```

Comment:
This is inlinable code -- we should not assume any specific malloc behavior that doesn't directly follow from its specs, even on Darwin. 😞 

We could round `realMinimumCapacity` up to the nearest 16-byte boundary before we call alloc, though!